sdk/key_vault/src/key.rs

```diff
@@ -0,0 +1,334 @@
+use std::fmt::{Debug, Display};
+
+use azure_core::TokenCredential;
+use chrono::serde::ts_seconds_option;
+use chrono::{DateTime, Utc};
+use getset::Getters;
+use serde::Deserialize;
+use serde_json::{Map, Value};
+use url::Url;
+
+use crate::client::API_VERSION;
+use crate::{KeyVaultClient, KeyVaultError};
+
+/// A KeyBundle consisting of a WebKey plus its attributes.
+#[derive(Debug, Deserialize, Getters)]
+#[getset(get = "pub")]
+pub struct KeyBundle {
+    /// The key management attributes.
+    attributes: KeyAttributes,
+    /// The Json web key.
+    key: JsonWebKey,
+    /// True if the key's lifetime is managed by key vault. If this is a key backing a certificate, then managed will be true.
+    managed: Option<bool>,
+    /// Application specific metadata in the form of key-value pairs.
+    tags: Option<Map<String, Value>>,
+}
+
+#[derive(Debug, Deserialize, Getters)]
+#[getset(get = "pub")]
+#[serde(rename_all = "camelCase")]
+pub struct KeyAttributes {
+    /// Creation time in UTC.
+    #[serde(with = "ts_seconds_option", default)]
+    created: Option<DateTime<Utc>>,
+    /// Determines whether the object is enabled.
+    enabled: Option<bool>,
+    /// Expiry date in UTC.
+    #[serde(with = "ts_seconds_option", default)]
+    exp: Option<DateTime<Utc>>,
+    /// Not before date in UTC.
+    #[serde(with = "ts_seconds_option", default)]
+    nbf: Option<DateTime<Utc>>,
+    /// softDelete data retention days. Value should be >=7 and <=90 when softDelete enabled, otherwise 0.
+    recoverable_days: Option<u8>,
+    /// Reflects the deletion recovery level currently in effect for keys in the current vault. If it contains 'Purgeable' the key can be permanently deleted by a privileged user; otherwise, only the system can purge the key, at the end of the retention interval.
+    recovery_level: Option<String>,
+    /// Last updated time in UTC.
+    #[serde(with = "ts_seconds_option", default)]
+    updated: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, Deserialize, Getters)]
+#[getset(get = "pub")]
+pub struct JsonWebKey {
+    /// Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+    crv: Option<String>,
+    /// RSA private exponent, or the D component of an EC private key.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    d: Option<String>,
```

Comment:
Across our other languages we base64url-decode these, such that the field types are already optional buffers, e.g. `Option<Vec<u8>>`. Fields will have different sizes for different key types / key sizes.